File: 728.Self-Dividing-Numbers/selfDividingNumbers.py
class Solution:
    def selfDividingNumbers(self, left, right):
        """
        :type left: int
        :type right: int
        :rtype: List[int]
        """
        result = []
        for i in range(left,right+1):
            if self.helper(i):#调用同类函数，需要用self调用
                result.append(i)
                
        return result
    
    def helper(self,numb):
        digits = list(str(numb))
        
        for i in digits:
            if i =='0' or numb % int(i) != 0:
                return False
        return True
        

File: 728.Self-Dividing-Numbers/selfDividingNumbers.cpp
/**
 * 辅助函数
 */
class Solution {
public:
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int> result;
        for(int i=left;i<=right;i++)
            if (isDividingNumbers(i))
                result.push_back(i);
        
        return result;
    }
    
    bool isDividingNumbers(int n){
        int n_orig = n;
        while(n>0){
            int digit = n % 10;
            if(!digit || n_orig % digit != 0)
                return false;
            n = n / 10;
        }
        return true;
    }
};
